apps/src/acemode/annotationList.js

```diff
@@ -12,22 +12,12 @@ function updateGutter() {
   if (!aceSession) {
     return;
   }
-  aceSession.setAnnotations(annotations);
 
   if (dropletEditor) {
-    // TODO: connect to missing hover event to show text
-    // TODO: differentiate between error and warning (type is always warning)
-
-    // Reset all decorations:
-    dropletEditor.gutterDecorations = {};
-    dropletEditor.redrawMain();
-
-    // Add each annotation as a gutter decoration:
-    for (var i = 0; i < annotations.length; i++) {
-      dropletEditor.addGutterDecoration(
-          annotations[i].row,
-          'droplet-' + annotations[i].type);
-    }
+    dropletEditor.setAnnotations(annotations);
```

Comment:
Might be worth a short comment here explaining that when we call dropletEditor.setAnnotations it calls aceSession.setAnnotations under the hood, otherwise one might assume we miss out on ace annotations in this path.